### PR TITLE
History reader

### PR DESCRIPTION
**Pull request by Udumft**
### Description
Adding history reader feature to improve debugging capabilities and to support feature parity between platforms.

### Implementation
Implementation is based on NN.HistoryReader which is designed to read `pbf` history files. Added several convenience wrapper methods to access contents as well as convenience initializers to simplify history replay setup, including free drive and AG scenarios.

### CODE REVIEW

File: Sources/MapboxCoreNavigation/HistoryFileReader.swift [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+public struct HistoryFileDump {
+    /// Initial `IndexedRouteResponse` that was set to the Navigator.
+    ///
+    /// Can be `nil` if current file is a free drive recording or if history recording was started after such event. In latter case this property may contain another `IndexedRouteResponse` which was, for example, set as a result of a reroute event.
+    public fileprivate(set) var initialRoute: IndexedRouteResponse?
+    /// Stream of location updates.
+    public fileprivate(set) var rawLocations: [CLLocation] = []
+}
+
+/// Provides event-by-event access to history files contents.
```
Comment (1ec5) [inline Sources/MapboxCoreNavigation/HistoryFileReader.swift:14]: Either this class comment or one of the method comments below should mention that `HistoryRecording.stopRecordingHistory(writingFileWith:)` writes a file to `HistoryRecording.historyDirectoryURL` that this class can read.

File: Sources/MapboxCoreNavigation/HistoryFileReader.swift [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+///
+/// Supports `pbf.gz` files.
+public final class HistoryFileReader {
```
[the remark this answers is not in the document]
Comment (Udumft) [reply Sources/MapboxCoreNavigation/HistoryFileReader.swift]: Adapted the reader to conform to `Sequence`.  Please check how it looks now

File: Sources/MapboxCoreNavigation/HistoryFileReader.swift [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+            guard let event = record.setRoute,
+                  let routeResponse = process(setRoute: event) else { break }
+            return HistorySetRoute(timestamp: timestamp,
+                                   routeResponse: routeResponse)
+        case .updateLocation:
+            guard let event = record.updateLocation else { break }
+            return HistoryUpdateLocation(timestamp: timestamp,
+                                         location: process(updateLocation: event))
+        default:
+            break
```
[the remark this answers is not in the document]
Comment (Udumft) [reply Sources/MapboxCoreNavigation/HistoryFileReader.swift]: Added them as internal objects. They make next to 0 sense to users. We are not exposing navigation status publicly in `RouteController` so we shouldn't do it here too, and push history event parameters are cryptic even to myself.
I think current approach will allow to introduce new types of events if they arrive later.

File: Sources/MapboxCoreNavigation/HistoryEvent.swift [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import Foundation
+import CoreLocation
+
+/// Base class for history events produced by `HistoryFileReader`
+public class HistoryEvent {
```
[the remark this answers is not in the document]
Comment (Udumft) [reply Sources/MapboxCoreNavigation/HistoryEvent.swift]: Reworked into protocol and different event types. Please see

**Pull request merged**